src/Language/Haskell/TH/Instances.hs

```diff
@@ -140,6 +154,31 @@ instance Fail.MonadFail Q where
 # endif
 #endif
 
+#if !MIN_VERSION_template_haskell(2,17,0)
+instance Semi.Semigroup a => Semi.Semigroup (Q a) where
+  (<>) = liftA2 (Semi.<>)
+
+instance Monoid a => Monoid (Q a) where
+  mempty = return mempty
+#if MIN_VERSION_base(4,11,0)
+  mappend = (Semi.<>)
```

Comment:
Can't we just use the default definition of `mappend` on `base-4.11+`? In other words, wouldn't the following work just as well?

```hs
instance Monoid a => Monoid (Q a) where
  ...
#if !(MIN_VERSION_base(4,11,0)
  mappend = liftA2 mappend
#endif
```

This accomplishes the same thing, but with less code.